src/helpers.rs

```diff
@@ -480,6 +480,23 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         self.eval_context_mut().memory.write_bytes(scalar, bytes.iter().copied().chain(iter::once(0u8)))?;
         Ok(true)
     }
+
+    fn alloc_os_str_as_c_str(
+        &mut self,
+        os_str: &OsStr,
+        memkind: MemoryKind<MiriMemoryKind>
+    ) -> Pointer<Tag> {
+        let bytes = os_str_to_bytes(os_str).unwrap();
+        let null_terminator = if cfg!(target_os = "windows") { 2 } else { 1 };
```

Comment:
This is not right -- you have to test the OS of the interpreted program; what you are doing here is testing the OS Miri runs on.

But also this is wrong; the function is explicitly writing as a *C String* and those are 1-byte-per-character on all platforms.